com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs

```diff
@@ -1101,20 +1116,30 @@ internal static void DisplayUnsupportedMessage(string msg)
 #endif
         }
 
-        internal static void DisplayUnsupportedAPIMessage(string graphicAPI = null)
+        internal static string GetUnsupportedAPIMessage(string graphicAPI)
         {
             // If we are in the editor they are many possible targets that does not matches the current OS so we use the active build target instead
 #if UNITY_EDITOR
             var buildTarget = UnityEditor.EditorUserBuildSettings.activeBuildTarget;
             string currentPlatform = buildTarget.ToString();
-            graphicAPI = graphicAPI ?? UnityEditor.PlayerSettings.GetGraphicsAPIs(buildTarget).First().ToString();
 #else
             string currentPlatform = SystemInfo.operatingSystem;
-            graphicAPI = graphicAPI ?? SystemInfo.graphicsDeviceType.ToString();
 #endif
 
-            string msg = "Platform " + currentPlatform + " with device " + graphicAPI + " is not supported with High Definition Render Pipeline, no rendering will occur";
-            DisplayUnsupportedMessage(msg);
+            string msg = "Platform " + currentPlatform + " with graphic API " + graphicAPI + " is not supported with HDRP";
+
+            // Display more information to the users when it should have use Metal instead of OpenGL
+            if (graphicAPI.StartsWith("OpenGL"))
+            {
+                if (SystemInfo.operatingSystem.StartsWith("Mac"))
+                    msg += ", use the Metal graphic API instead";
+                else if (SystemInfo.operatingSystem.StartsWith("Windows"))
+                    msg += ", use the Vulkan graphic API instead";
+            }
+
+            msg += ".\nPlease change the platform/device to a compatible one or remove incompatible graphics APIs in Project Settings > Player > Other Settings.";
```

Comment:
.\nChange the platform/device to a compatible one or remove incompatible graphics APIs. To do this, go to Project Settings > Player > Other Settings and modify the respective Graphics APIs for <platform> list.